src/System.Device.Gpio/System/Device/Gpio/GpioPin.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Represents a general-purpose I/O (GPIO) pin.
+    /// </summary>
+    public sealed class Gpio​Pin : IDisposable
```

Comment:
I'm curious about this. If we make this abstract there we could potentially create better implementation for i.e. RaspberryPi3Driver which can directly write to memory rather than going through controller. Leaving as is might be ok but we should consider being able to use this in LedMatrix code which could benefit from that design change

Reply:
Why not, we can definitely make it abstract for other usage. Let me think of this. The key question then for the abstract is passing or not the controller and having a default controller based implementation or not

Reply:
It would be slightly more expensive, since it would require another virtual call.

Reply:
if most derived GpioPin is used directly JIT will devirtualize that and make it equally fast. The only time you'd pay for virtual cost is if you used GpioPin directly. For most usages it won't matter since hardware/OS will limit us but for where it matters it will be faster to do it this way (I think)

Reply:
As part of this proposal we should run some benchmarks comparing old approach vs new.